src/main/java/appeng/client/gui/widgets/MEGuiTextField.java

```diff
@@ -86,4 +94,69 @@ public boolean isMouseIn( final int xCoord, final int yCoord )
 
 		return withinXRange && withinYRange;
 	}
+
+	public void selectAll()
+	{
+		this.setCursorPosition( 0 );
+		this.setSelectionPos( this.getMaxStringLength() );
+	}
+
+	public void setSelectionColor( int color )
+	{
+		this.selectionColor = color;
+	}
+
+	@Override
+	public void drawSelectionBox( int startX, int startY, int endX, int endY )
+	{
+		if( startX < endX )
+		{
+			int i = startX;
+			startX = endX;
+			endX = i;
+		}
+
+		startX += 1;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "startX". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1226)